### PR TITLE
airoha: convert to MFD implementation

### PR DESCRIPTION
**Pull request by Ansuel**

### CODE REVIEW

File: target/linux/airoha/files/drivers/mfd/airoha-en7581-mfd.c [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	mfd = devm_kzalloc(dev, sizeof(*mfd), GFP_KERNEL);
+	if (!mfd)
+		return -ENOMEM;
+	platform_set_drvdata(pdev, mfd);
+
+	mfd->base = devm_platform_ioremap_resource(pdev, 0);
+	if (IS_ERR(mfd->base))
+		return PTR_ERR(mfd->base);
+
+	mfd->chip_scu = syscon_regmap_lookup_by_phandle(dev->of_node,
```
Comment (LorenzoBianconi) [inline target/linux/airoha/files/drivers/mfd/airoha-en7581-mfd.c]: @Ansuel I guess we can use compatible string instead

File: target/linux/airoha/files/drivers/pwm/pwm-airoha.c [hunk 2]
```diff
@@ -7,20 +7,15 @@
...
 #include <linux/module.h>
 #include <linux/of.h>
 #include <linux/platform_device.h>
 #include <linux/pwm.h>
-#include <linux/regmap.h>
 #include <linux/gpio.h>
 #include <linux/bitops.h>
 #include <asm/div64.h>
 
-#define REG_SGPIO_CFG			0x0024
```
Comment (LorenzoBianconi) [inline target/linux/airoha/files/drivers/pwm/pwm-airoha.c]: @Ansuel are they not used?

File: target/linux/airoha/files/drivers/pwm/pwm-airoha.c [hunk 3]
```diff
@@ -386,16 +388,16 @@ static const struct pwm_ops airoha_pwm_ops = {
...
 	struct airoha_pwm *pc;
 
 	pc = devm_kzalloc(&pdev->dev, sizeof(*pc), GFP_KERNEL);
 	if (!pc)
 		return -ENOMEM;
 
-	pc->base = syscon_node_to_regmap(parent->of_node);
-	if (IS_ERR(pc->base))
-		return PTR_ERR(pc->base);
+	pc->sgpio_cfg = mfd->base + 0x24;
```
Comment (LorenzoBianconi) [inline target/linux/airoha/files/drivers/pwm/pwm-airoha.c]: @Ansuel no magic numnbers

File: target/linux/airoha/files/drivers/mfd/airoha-en7581-mfd.c (hunk 1 above)
Comment (Ansuel) [reply target/linux/airoha/files/drivers/mfd/airoha-en7581-mfd.c]: Main problem is that using absolute compatible is problematic for yaml schema as we don't have a way to reference them... IMHO phandle is better

File: target/linux/airoha/files/drivers/pinctrl/pinctrl-airoha.c [hunk 4]
```diff
@@ -2882,6 +2882,7 @@ static int airoha_pinctrl_add_gpiochip(struct airoha_pinctrl *pinctrl,
 static int airoha_pinctrl_probe(struct platform_device *pdev)
 {
 	struct device *dev = &pdev->dev;
+	struct airoha_mfd *mfd = dev_get_drvdata(dev->parent);
 	struct airoha_pinctrl *pinctrl;
```
Comment (LorenzoBianconi) [inline target/linux/airoha/files/drivers/pinctrl/pinctrl-airoha.c:2886]: @Ansuel rct whenever possible

File: target/linux/airoha/files/include/linux/mfd/airoha-en7581-mfd.h [hunk 5]
```diff
@@ -0,0 +1,10 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+#ifndef _LINUX_INCLUDE_MFD_AIROHA_EN7581_MFD_H_
+#define _LINUX_INCLUDE_MFD_AIROHA_EN7581_MFD_H_
+
+struct airoha_mfd {
+	void __iomem *base;
+	struct regmap *chip_scu;
```
Comment (LorenzoBianconi) [inline target/linux/airoha/files/include/linux/mfd/airoha-en7581-mfd.h]: @Ansuel this is not necessary I would say, we can do the lookup in the pinctrl driver

**Pull request merged**